tests/test_data/test_filter.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import polars as pl
+
+from ..context import nhs
+
+filter_sa1_regions = nhs.data.filter.filter_sa1_regions
+
+class TestFilterSa1RegionCodes:
+  
+    # Fixture to create a sample LazyFrame
+    @pytest.fixture
+    def sample_lazyframe(self):
+        data = {
+            "SA1_CODE_2021": ["123456", "789012", "345678", "901234", "567890"],
+            "value": [10, 20, 30, 40, 50]
+        }
+        return pl.DataFrame(data).lazy()
```

Comment:
Just FYI: you should be able to do `pl.LazyFrame({"SA1_CODE_"...})` instead of doing `pl.DataFrame(...).lazy()`. Don't need to change this, the comment is more for next time